### ISSUE TITLE
Regression or issue persistence: "command not found: podman" after install on m1 mac

### ISSUE THREAD

**Issue by uogbuji** [containers/podman#18676]
### Issue Description

"command not found: podman" after install on m1 mac, already reported in #15542. OP there closed it without clear resolution & issue persists.

### Steps to reproduce the issue

On an m1 mac (MPB 16" 2021 running Ventura 13.3.1):

Install podman desktop, then follow the "Podman is not installed on this system, would you like to install Podman 4.5.0?" prompt. Complete the installation wizard; select to install for all users on this machine.

Go to terminal and type `podman info`; type `rehash` and try again

### Describe the results you received

```
zsh: command not found: podman
```

### Describe the results you expected

Successful run of the command

### podman info output

```yaml
Client:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.19
Git Commit:   75e3c12579d391b81d871fd1cded6cf0d043550a
Built:        Fri Apr 14 10:04:39 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.20.2
Built:        Fri Apr 14 09:41:12 2023
OS/Arch:      linux/arm64
(loomi) ➜  etc /opt/podman/bin/podman info   
host:
  arch: arm64
  buildahVersion: 1.30.0
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.7-2.fc38.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 95.07
    systemPercent: 3.63
    userPercent: 1.31
  cpus: 1
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: coreos
    version: "38"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
    uidmap:
    - container_id: 0
      host_id: 501
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
  kernel: 6.2.15-300.fc38.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 1241636864
  memTotal: 2049007616
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.8.4-1.fc38.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.4
      commit: 5a8fa99a5e41facba2eda4af12fa26313918805b
      rundir: /run/user/501/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/501/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-12.fc38.aarch64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 0h 8m 10.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 106769133568
  graphRootUsed: 2546946048
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/501/containers
  transientStore: false
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 4.5.0
  Built: 1681486872
  BuiltTime: Fri Apr 14 09:41:12 2023
  GitCommit: ""
  GoVersion: go1.20.2
  Os: linux
  OsArch: linux/arm64
  Version: 4.5.0
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

M1 mac
Physical

### Additional information

This was already reported in #15542. OP there closed it without clear resolution.

Podman command is indeed installed at: `/opt/podman/bin` but this is not in path. No update by installer to `~/.zshenv`, `~/.zshrc` or `~/.zprofile`. In fact, doing `grep podman ~/.*` shows nothing except for the `~/.zsh_history` entries.

I'm actually following [this Red Hat guide here](https://developers.redhat.com/articles/2023/02/14/remote-container-development-vs-code-and-podman#install_podman_desktop_on_the_local_machine), and if I simulate what it prescribes:

```
/opt/podman/bin/podman version
```

I get:

```
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman socket: Get "http://d/v4.5.0/libpod/_ping": dial unix ///var/folders/61/pb7hvq8x34jc9v6cc017kx2r0000gn/T/podman-run--1/podman/podman.sock: connect: no such file or directory
```

Running 

```
/opt/podman/bin/podman machine init
/opt/podman/bin/podman machine start
```

Fixes it, as expected, but those steps are not in the above guide. Resulting version output:

```
Client:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.19
Git Commit:   75e3c12579d391b81d871fd1cded6cf0d043550a
Built:        Fri Apr 14 10:04:39 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.20.2
Built:        Fri Apr 14 09:41:12 2023
OS/Arch:      linux/arm64
```

### SIDE NOTE:

This link in the bug guidance template is wrong: https://github.com/containers/podman/README.md. It should be https://github.com/containers/podman/blob/main/README.md

**Comment by Luap99** [comment]
Did you restart your shell after the install?

**Comment by Luap99** [comment]
https://github.com/containers/podman/blob/74d804d1ca02bf1ae1a082bef14fa77979ce9cf6/contrib/pkginstaller/scripts/postinstall#L5

**Comment by uogbuji** [comment]
Above comment triggered a thought: OP in #15542 mentioned an update to `.bash_profile`. But Macs have switched to zsh by default for a while now. I do not have bash, nor any of its dotfiles, and certainly not `.bash_profile`. Could the issue be that the podman installer isn't set up for the default zsh-only?

**Comment by Luap99** [comment]
We do not touch any dot file, the installer writes the PATH to /etc/paths.d/podman-pkg. I would expect that to be picked up by your shell on a restart. I do not own a mac so I cannot test it.

**Comment by uogbuji** [comment]
OK I did launch a new shell, and that works. `/opt/podman/bin` is in the `PATH`. So you were right, I needed to restart the shell, and running `rehash` was not enough, but that's pretty annoying. I guess it's more an issue with the Mac OS setup than with podman, though, so I'll close this. I just recommend making sure your various guides are updated to b clear on this point, because old-school UNIX/Linux heads like me are not going to be thinking they should need to restart their shells without very good reason.

I'm also glad we have this ticket with a little more context for the next person who comes along.
